### ISSUE TITLE
How can I do T-SNE analysis on the classifier of the detector?

### ISSUE THREAD

**Issue by hjfdsssdg** [open-mmlab/mmdetection#9833]
How do I do T-SNE analysis on the classifier of the detector? Is there any corresponding reference material based on mmdet? For example, to do T-SNE on the output of the FC layer classifier of Faster RCNN

**Comment by abinav719** [comment]
I had a similar problem as well and i am all ears. I saved the logits by parsing it through the scores. Then did a t-sne analysis on the last layer logits. But this works as a post processing and not during the training process.

**Comment by hjfdsssdg** [comment]
@abinav719 I think T_sne analysis can be done both during model training and testing, it mainly captures the output of feature weights and labels at forward. My current confusion is that I don't know how to get the correct classification label during testing via mmdet

**Comment by abinav719** [comment]
During testing we cannot pass labels. So i got the bboxes and did reassociation to find which bboxes corresponded to the right classification label because u have true labels from annotations in xml for pascal voc

**Comment by 873552584** [comment]
> 在测试期间，我们无法通过标签。所以我得到了 bboxes 并重新关联以找到哪些 bboxes 对应于正确的分类标签，因为你有来自 xml for pascal voc 注释的真实标签

Can you provide the t-SNE visualization code about mmdet, please? Thank you very much, and I am willing to communicate with you. I am very much looking forward to your reply.

**Comment by FindOne1** [comment]
> > 在测试期间，我们无法通过标签。所以我得到了 bboxes 并重新关联以找到哪些 bboxes 对应于正确的分类标签，因为你有来自 xml for pascal voc 注释的真实标签
> 
> Can you provide the t-SNE visualization code about mmdet, please? Thank you very much, and I am willing to communicate with you. I am very much looking forward to your reply.

Have you been able to run the t-SNE visualization code? I want a copy of the code, too.
